### PR TITLE
Skill - ArdentSacrifice

### PR DESCRIPTION
**Pull request by JesterWizard**
Sacrifice 10 HP to restore 10 HP to an allied adjacent unit.

### CODE REVIEW

**Comment by MokhaLeee** [comment]
conflicts exists, also better to reduce text index consumption. The text index is not enough

File: Wizardry/Core/SkillSys/MiscSkillEffects/MenuSkills/ArdentSacrifice.c [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    struct Unit * unit_tar = GetUnit(gActionData.targetIndex);
+
+    CallMapAnim_HealExt(
+        parent,
+        unit_tar,
+        SKILL_EFF1(SID_ArdentSacrifice),
+        ActionArdentSacrifice_CallBack1,
+        NULL);
+
+    SetUnitHp(gActiveUnit, GetUnitCurrentHp(gActiveUnit) - SKILL_EFF0(SID_ArdentSacrifice));
```
Comment (MokhaLeee) [inline Wizardry/Core/SkillSys/MiscSkillEffects/MenuSkills/ArdentSacrifice.c:93]: also no need this

**Comment by MokhaLeee** [comment]
Remove `MSG_MenuSkill_ArdentSacrifice_Target` and `MSG_MenuSkill_ArdentSacrifice_FRtext`

**Comment by JesterWizard** [comment]
> Remove `MSG_MenuSkill_ArdentSacrifice_Target` and `MSG_MenuSkill_ArdentSacrifice_FRtext`

I'm resistant to removing ``MSG_MenuSkill_ArdentSacrifice_Target``. It helps to know what the skill is going to do when you select a unit. 

Fair enough on ``MSG_MenuSkill_ArdentSacrifice_FRtext``. In retrospect, it doesn't seem like it'll ever be seen.

**Comment by JesterWizard** [comment]
> conflicts exists, also better to reduce text index consumption. The text index is not enough

Is the text index set per skill category or for everything? In either case, it'll probably need to be expanded then, because there's 10 more menu skills to go. https://docs.google.com/spreadsheets/d/1zd99uevX953OZQiA_oJU9R3hqFao5mrW78wMHKPco-U/edit?gid=681794002#gid=681794002

**Comment by MokhaLeee** [comment]
> It helps to know what the skill is going to do when you select a unit.

well since the msg index has go over more than 0x1000， it is more important to save the index

**Comment by MokhaLeee** [comment]
There has been one common rtext msg and I think no need to add an unique rtext.

**Comment by JesterWizard** [comment]
> > It helps to know what the skill is going to do when you select a unit.
> 
> well since the msg index has go over more than 0x1000， it is more important to save the index

I won't remove the subtitle, but I'm open to changing the text to a pre-existing message if storage is at a premium.

File: Wizardry/Core/SkillSys/MiscSkillEffects/MenuSkills/ArdentSacrifice.c [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    struct Unit * unit_tar = GetUnit(gActionData.targetIndex);
+
+    CallMapAnim_HealExt(
+        parent,
+        unit_tar,
+        SKILL_EFF1(SID_ArdentSacrifice),
+        ActionArdentSacrifice_CallBack1,
+        NULL);
+
+    CallMapAnim_Hurt(parent, gActiveUnit, SKILL_EFF0(SID_ArdentSacrifice));
```
Comment (MokhaLeee) [inline Wizardry/Core/SkillSys/MiscSkillEffects/MenuSkills/ArdentSacrifice.c:92]: Can they cooperate properly for hurt and heal at the same time?

File: include/skill-system.h [hunk 3]
```diff
@@ -319,3 +321,4 @@ bool Action_GoddessDance(ProcPtr parent);
 bool Action_Stride(ProcPtr parent);
 bool Action_Swarp(ProcPtr parent);
 bool Action_Blacksmith(ProcPtr parent);
+bool Action_ArdentSacrifice(ProcPtr proc);
```
Comment (MokhaLeee) [inline include/skill-system.h:324]: Add blank line at end of the file

File: Wizardry/Core/SkillSys/MiscSkillEffects/MenuSkills/ArdentSacrifice.c (hunk 2 above)
Comment (JesterWizard) [reply Wizardry/Core/SkillSys/MiscSkillEffects/MenuSkills/ArdentSacrifice.c:92]: Well yes and no. They both happen sequentially (though hurt happens first for some reason) but it seems the skill holder is temporarily hidden when the target unit is healing. Unsure if there's a function to stop that.

Comment (MokhaLeee) [reply Wizardry/Core/SkillSys/MiscSkillEffects/MenuSkills/ArdentSacrifice.c:92]: I think that's maybe because they cannot exec at the same time

Comment (JesterWizard) [reply Wizardry/Core/SkillSys/MiscSkillEffects/MenuSkills/ArdentSacrifice.c:92]: Then I guess I need to look into blocks and stuff, using swarp as a reference point. I'll get on that later.

**Pull request closed without merging**